src/main/java/org/medicmobile/webapp/mobile/Utils.java

```diff
@@ -31,7 +32,9 @@ static boolean isUrlRelated(String appUrl, Uri uriToTest) {
 		// android.net.Uri doesn't give us a host for URLs like blob:https://some-project.dev.medicmobile.org/abc-123
 		// so we might as well just regex the URL string
 		if (uriToTest != null) {
-			return isUrlRelated(appUrl, uriToTest.toString());
+			String uriToTestString = uriToTest.toString();
+			return isUrlRelated(appUrl, uriToTestString)
+					|| (uriToTestString.contains("redirect_uri="+ Uri.encode(appUrl)+Uri.encode("/medic/login/oidc")));
```

Comment:
I am a bit worried about unintended consequences of including this logic down here in the generic `isUrlRelated` function. This function also gets called via `isValidNavigationUrl` [when the app is closed](https://github.com/medic/cht-android/blob/8020cd4924a0b92390d8c1543a3e9186c5e20c11/src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java#L161) (as part of the logic to save the user's current url to open next time).  If the app gets closed during the middle of an OIDC login flow, I don't think we want to save that URL.

Reply:
thanks @jkuester all good for me with your improvements.